modules/gateway/peers.go

```diff
@@ -233,35 +233,26 @@ func (g *Gateway) Disconnect(addr modules.NetAddress) error {
 	return nil
 }
 
-// makeOutboundConnections tries to keep the Gateway well-connected. As long
-// as the Gateway is not well-connected, it tries to add random nodes as
-// peers. It sleeps when the Gateway becomes well-connected, or it has tried
-// more than 100 nodes.
-func (g *Gateway) makeOutboundConnections() {
+// peerManager tries to keep the Gateway well-connected. As long as the
+// Gateway is not well-connected, it tries to connect to random nodes.
+func (g *Gateway) peerManager() {
 	for {
-		for i := 0; i < 100; i++ {
-			id := g.mu.RLock()
-			numPeers := len(g.peers)
-			addr, err := g.randomNode()
-			g.mu.RUnlock(id)
-			if err != nil || numPeers >= wellConnectedThreshold {
-				break
-			}
-			err = g.Connect(addr)
-			// aggressively remove unresponsive nodes
-			if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
-				id = g.mu.Lock()
-				g.removeNode(addr)
-				g.mu.Unlock(id)
-			}
-		}
-		// request more nodes if necessary
+		// If we are well-connected, sleep in increments of five minutes until
+		// we are no longer well-connected.
 		id := g.mu.RLock()
-		numNodes := len(g.nodes)
-		addr, err := g.randomPeer()
+		numPeers := len(g.peers)
+		addr, err := g.randomNode()
 		g.mu.RUnlock(id)
-		if build.Release != "testing" && err == nil && numNodes < minNodeListLen {
-			g.RPC(addr, "ShareNodes", g.requestNodes)
+		if numPeers >= wellConnectedThreshold {
+			time.Sleep(5 * time.Minute)
+			continue
+		}
```

Comment:
the node isn't used in this branch, due to the `continue`.
It might be clearer to grab the node after the sleep, but that would require two readlocks.